measurements/toxicity/toxicity.py

```diff
@@ -0,0 +1,136 @@
+# Copyright 2020 The HuggingFace Evaluate Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+""" Toxicity detection measurement. """
+
+import datasets
+from transformers import AutoModelForSequenceClassification, AutoTokenizer, pipeline
+
+import evaluate
+
+
+logger = evaluate.logging.get_logger(__name__)
+
+
+_CITATION = """
+@inproceedings{vidgen2021lftw,
+  title={Learning from the Worst: Dynamically Generated Datasets to Improve Online Hate Detection},
+  author={Bertie Vidgen and Tristan Thrush and Zeerak Waseem and Douwe Kiela},
+  booktitle={ACL},
+  year={2021}
+}
+"""
+
+_DESCRIPTION = """\
+The toxicity measurement aims to quantify the toxicity of the input texts using a hate speech classification model trained for the task
+"""
+
+_KWARGS_DESCRIPTION = """
+Compute the toxicity of the input sentences.
+
+Args:
+    `predictions` (list of str): prediction/candidate sentences
+    `toxic_label` (optional): the toxic label that you want to detect, depending on the labels that the model has been trained on.
```

Comment:
and how would you aggregate the results across different labels?
e.g. if you have 
{'offensive': 0.65, 'hate': 0.98}, then what?

Reply:
Ah yes, aggregation would be a bit tricky. I think the Perspective API (as an example) reports these results back unaggregated across categories — there's an individual score for each category of 'identity hate', 'toxicity', 'sexism', 'racism', 'sexually explicit' etc. and they don't aggregate across categories. 

I assume the idea is that as an end user of a toxicity API you'd want to handle cases of sexually explicit content differently than identity-based hate, so the granularity is helpful/necessary. In this case an equivalent process would be to not aggregate when there are several types of toxicity specified, and report back per-toxicity-class (e.g. "toxic_labels" is a list instead of a str). What do you think?

Reply:
For now I can only find binary hate speech classification models on the Hub, so maybe we keep it like this for now?